File: mini_browser.go
// mini_browser.go — Go версия

package main

import (
	"flag"
	"fmt"
	"os"
	"os/exec"
	"runtime"
)

func main() {
	url := flag.String("url", "https://example.com", "URL для открытия")
	fullscreen := flag.Bool("fullscreen", true, "Полноэкранный режим")
	flag.Parse()

	fmt.Println("🌐 Mini-Browser (Go)")
	fmt.Printf("Открыт URL: %s\n", *url)
	fmt.Println("Для выхода нажмите Ctrl+C")

	var cmd *exec.Cmd

	switch runtime.GOOS {
	case "windows":
		if *fullscreen {
			cmd = exec.Command("cmd", "/c", "start", "/max", *url)
		} else {
			cmd = exec.Command("cmd", "/c", "start", *url)
		}
	case "darwin":
		cmd = exec.Command("open", *url)
		if *fullscreen {
			// macOS: используем AppleScript для полноэкранного режима
			script := fmt.Sprintf(`tell application "Safari" to activate
tell application "System Events"
	tell process "Safari"
		keystroke "f" using {command down, control down}
	end tell
end tell`)
			cmd = exec.Command("osascript", "-e", script)
		}
	default: // Linux
		if *fullscreen {
			cmd = exec.Command("google-chrome", "--kiosk", *url)
		} else {
			cmd = exec.Command("xdg-open", *url)
		}
	}

	cmd.Stdout = os.Stdout
	cmd.Stderr = os.Stderr
	if err := cmd.Run(); err != nil {
		fmt.Printf("❌ Ошибка: %v\n", err)
		os.Exit(1)
	}
}

File: mini_browser.php
<?php
// mini_browser.php — PHP версия

$url = "https://example.com";
$fullscreen = true;

// Парсим аргументы командной строки
for ($i = 1; $i < $argc; $i++) {
    if ($argv[$i] == "--url") {
        $url = $argv[++$i];
    } elseif ($argv[$i] == "--no-fullscreen") {
        $fullscreen = false;
    }
}

echo "🌐 Mini-Browser (PHP)\n";
echo "Открыт URL: $url\n";
echo "Для выхода закройте окно браузера\n";

$command = "";
if (strtoupper(substr(PHP_OS, 0, 3)) === 'WIN') {
    // Windows
    if ($fullscreen) {
        $command = "start /max $url";
    } else {
        $command = "start $url";
    }
} elseif (PHP_OS === 'Darwin') {
    // macOS
    $command = "open $url";
} else {
    // Linux
    if ($fullscreen) {
        $command = "google-chrome --kiosk $url &";
    } else {
        $command = "xdg-open $url &";
    }
}

exec($command);

// Ждём завершения (упрощённо)
sleep(3600);
?>
